gax-java/gax/src/main/java/com/google/api/gax/batching/BatcherImpl.java

```diff
@@ -241,12 +251,14 @@ public ApiFuture<ElementResultT> add(ElementT element) {
 
     SettableApiFuture<ElementResultT> result = SettableApiFuture.create();
     synchronized (elementLock) {
-      currentOpenBatch.add(element, result, throttledTimeMs);
-    }
+      if (batchingDescriptor.shouldFlush(
+          currentOpenBatch.resource.add(newResource), elementThreshold, bytesThreshold)) {
```

Comment:
This can be simplified to `resource.add(newResource).shouldFlush(elementThreshold, bytesThreshold))` once the `shouldFlush` method is moved to `BatchResource` or `DefaultBatchResource` if we decide to get rid of the interface.